concurrency.go

```diff
@@ -99,6 +100,11 @@ func Async6[A, B, C, D, E, F any](f func() (A, B, C, D, E, F)) <-chan Tuple6[A,
 
 // WaitFor runs periodically until a condition is validated.
 func WaitFor(condition func(i int) bool, maxDuration time.Duration, tick time.Duration) (int, time.Duration, bool) {
+	return WaitForWithContext(context.Background(), condition, maxDuration, tick)
+}
+
+// WaitForWithContext runs periodically until a condition is validated or context is canceled.
+func WaitForWithContext(ctx context.Context, condition func(i int) bool, maxDuration time.Duration, tick time.Duration) (int, time.Duration, bool) {
```

Comment:
I wonder if condition should receive the context. WDYT ?

Reply:
Oh that's a good point. I don't think so

Because the only need would be if the method we had would pass the context to 
f when iteratimg. The only need would be if the code in WaitForWithContext would play with the context.

Also, right now, if you need the pass a context to the condition you can do it with this

```go
myCtx := ctxThatCanExpire
condition := func(I int) {
   func(ctx context. Context) {
        // do something with context
   }(myCtx)
}

lo.WaitForWithContext(ctx, condition, d, t)
```

But now, I wrote it. Yes, maybe we could passing the context is easy, and it might simply might be ignored, as `i` right now


So you tell me, I can update the code

Reply:
pushed back, please review and resolve this thread